### ISSUE TITLE
updating the docker image

### ISSUE THREAD

**Issue by vijayrajah** [huginn/huginn#1431]
I have a docker based instance of huginn running on my server.

How do i update the image. I mean, If i do a "docker pull" will that remove the existing and initialize an new on?.. 

What is the official method of updating docker images.

**Comment by johnnyzen** [comment]
I would love to know this too :)

**Comment by dsander** [comment]
`docker pull` will only update your local huginn image. To update the container your need to run `docker stop`, `docker rm`, and `docker run` to create a new container based on the updated image. Keep in mind that this will remove the data inside the old huginn container, when you are not using a [linked database or exported the data using a volume](https://github.com/cantino/huginn/blob/master/docker/multi-process/README.md#usage). 
When you are using docker-compose this will update to the latest version:
```
docker-compose stop
docker-compose pull
docker-compose up -d
```

**Comment by vijayrajah** [comment]
Yes, Thanks for this @dsander .. I was running an instance which did not have mysql volume exported.. 
@johnnyzen 
This is what i did

- Take a backup -- docker export -o  ~/huginn.tar <container_name>

- Find the volume for the current huginn image -- use docker inspect <cointainer_name" and find the fs path for the container
- do an rsync of the volume to a directory on the docker_host (i used: rsynv -avp ..)
- stop the container -- docker stop
- remove the container -- docker rm
- get the latest code -- docker pull
- Instantiate a new container with the latest code -- docker exec 
       - I passed several environment variables like

`-e DO_NOT_SEED=1 \
-e APP_SECRET_TOKEN="<>" \
-e TWITTER_OAUTH_KEY=<> \
-e TWITTER_OAUTH_SECRET=<> \
-e TIMEZONE="<>" \`


Hope this helps

**Comment by johnnyzen** [comment]
thanks @vijayrajah thats very kind of you to write that up :)

**Comment by christopherdurr** [comment]
I believe that this issue has been resolved and can be closed -- not sure who closes it though.

**Comment by jjeffers** [comment]
Closing this per @christopherdurr 's comment.
